plugins/palo_alto_cortex_xdr/plugin.spec.yaml

```diff
@@ -290,3 +319,13 @@ triggers:
         description: Incident
         type: object
         required: false
+  get_alerts:
+    title: Get Alerts
+    description: Get Alerts
+    output:
+      alert:
+        title: Alert
+        description: Alert
+        type: object
```

Comment:
Is there a common schema output where we can create a [custom type](https://docs.rapid7.com/insightconnect/plugin-spec/#the-plugin-specification) for, rather than a generic object? For InsightConnect, users can't access the individual keys by their key names in the product since we don't define the schema for them. Instead they would need to use JSON features of plugins to try and parse them.

There is a helpful script that can convert a JSON response to a custom type. https://github.com/rapid7/insightconnect-plugins/blob/master/tools/json_to_spec.py

e.g.

```
types:
  alert:
    content1:
       type: string
     content2:
       type: string
....

get_alerts:    
  title: Get Alerts    
  description: Get Alerts    
  output:      
     alert:        
     title: Alert        
     description: Alert        
     type: alert
```

Reply:
That's an amazing tool! I have replaced the object type in get alerts and get incidents with a custom alert & incident type. 

I have a question though, for some of the fields their value in the only example of an alert I have, are None. So when you run the alert through that tool it adds `type: None` in the plugin spec, for some fields it's easy enough to guess if they are string or int or whatever. 

My question is, if I get that type wrong in the plugin spec, and we get an alert where one of the fields is in the spec as a string, but we get an alert where that field is a list, for example. What would the knock-on effects be?